org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/DiagnosticResolution.java

```diff
@@ -61,28 +85,31 @@ public String getLabel() {
 	}
 
 	public WorkspaceEdit apply() {
-		DiagnosticModificationContext modificationContext = factory.createModificationContext();
 		try {
 			XtextResource resource = options.getResource();
 			URI uri = resource.getURI();
-			ILanguageServerAccess access = options.getLanguageServerAccess();
-
-			WorkspaceEdit edit = new WorkspaceEdit();
-			ChangeConverter2 changeConverter = modificationContext.getConverterFactory().create(edit, access);
-
-			ResourceSet resourceSet = access.newLiveScopeResourceSet(uri);
-			XtextResource tmpResource = (XtextResource) resourceSet.getResource(uri, true);
 
 			CodeActionParams params = options.getCodeActionParams();
-
-			int offset = options.getDocument().getOffSet(params.getRange().getStart());
+			Document document = options.getDocument();
+			int offset = document.getOffSet(params.getRange().getStart());
 			EObjectAtOffsetHelper helper = new EObjectAtOffsetHelper();
+			ILanguageServerAccess access = options.getLanguageServerAccess();
+			ResourceSet resourceSet = access.newLiveScopeResourceSet(uri);
+			XtextResource tmpResource = (XtextResource) resourceSet.getResource(uri, true);
 			EObject obj = helper.resolveContainedElementAt(tmpResource, offset);
 
-			IChangeSerializer serializer = modificationContext.getSerializer();
-			serializer.addModification(obj, modification);
-			serializer.applyModifications(changeConverter);
-
+			WorkspaceEdit edit = new WorkspaceEdit();
+			if (modification != null) {
+				DiagnosticModificationContext modificationContext = factory.createModificationContext();
+				ChangeConverter2 changeConverter = modificationContext.getConverterFactory().create(edit, access);
+				IChangeSerializer serializer = modificationContext.getSerializer();
+				serializer.addModification(obj, modification);
+				serializer.applyModifications(changeConverter);
+			} else {
+				TextEditAcceptor textEditAcceptor = new TextEditAcceptor(edit, access);
+				Diagnostic diagnostic = params.getContext().getDiagnostics().get(0);
```

Comment:
I'm a little scared about the (`.get(0)`) here. This is probably incorrect.

Reply:
The code in [org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/codeActions/QuickFixCodeActionService.java](https://github.com/eclipse/xtext-core/blob/942def122c1b49feac1713de2edeae514b0b1b7c/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/codeActions/QuickFixCodeActionService.java#L69-71) makes sure that we have a singleton list, with exactly one element, so as the code currently is, I do not think this is a mistake, as it relies on an invariant, but I also dislike it. I left the code as it is because I am not sure what should be better than this.
Should we add a precondition check that tests that the list has indeed one and only one argument? Would you prefer that?

Reply:
Yes, please use something like com.google.common.collect.Iterables.getOnlyElement(Iterable<T>) to document and assert that invariant in code.

Reply:
Nice, I was not aware of that Guava API. I have opted for using what they document as the variant for Java 8 using streams.